Test fixes

jedwards4b opened the pull request:
Fix issues with ERS, ERR, ERP tests, add copy of env_mach_pes.xml for PFS test.  
Update cime_developer test list to catch errors that were missed in the commit that this 
commit fixes.  

Test suite: scripts_regression_tests.py , cesm yellowstone prealpha
Test baseline: cesm2_0_alpha06a
Test namelist changes: 
Test status: bit for bit

Fixes #1066 
Fixes #1067 

User interface changes?: 

Code review:

gold2718 commented:
Is there a reason you haven't run the cime_developer tests? Or is that just something you need to add to your PR doc?

jedwards4b commented:
cime_developer is Z_FullSystemTest in scripts_regression_tests.py

jgfouca commented on utils/python/CIME/check_lockedfiles.py:
```diff
@@ -41,7 +42,7 @@ def restore(filename, caseroot=None, newname=None):
     caseroot = os.getcwd() if caseroot is None else caseroot
     newname = filename if newname is None else newname
-    shutil.copyfile(os.path.join(caseroot, LOCKED_DIR, newname), os.path.join(caseroot, filename))
+    shutil.copyfile(os.path.join(caseroot, LOCKED_DIR, filename), os.path.join(caseroot, newname))
```
Thanks, wondering how any tests passed with this mistake :/

jedwards4b commented:
I changed the cime_developer test, went back and ran the changed test on the broken commit and made sure that it failed.

jgfouca commented:
@gold2718 I specifically asked JimE to change the tests so that this kind of thing will be found in the future and he followed through. It was a pretty small change to do this, so you may not have noticed it; it's in update_acme_tests.

gold2718 commented:
@jgfouca, yeah, I was in the middle of looking at the changes when you merged the PR.
Can we at least agree to run cime_developer any time anything in SystemTests is changed?
That might have (should have?) prevented this whole issue.

jgfouca commented:
@gold2718 oh, sry to interrupt your review. We can change the PR after the merge by the way, so you can still request changes. You can also add yourself as a review so I'll know to wait.

I believe all the tests were run when the broken PR went in but they didn't catch it, but they should in the future.

jedwards4b commented:
cime_developer IS the same as scripts_regression_tests.py Z_FullSystemTest.

gold2718 commented:
@jedwards4b, can you update the testing page (https://github.com/ESMCI/cime/wiki/Testing) to explain all this? The current description seems to clash with what you just said (and Z_FullSystemTest is not even mentioned).
TIA

jedwards4b commented:
It says this 
```
   ./scripts_regression_tests.py

This will run X and A system tests (the same as what you get from running `./create_test cime_developer`)
```

jedwards4b commented:
I don't think that it really needs the detail that this is called Z_FullSystemTest does it???

gold2718 commented:
Okay, so this is slowly percolating through an apparently thicker-than-usual skull.
I think you are saying that the doc is correct but wasn't before this latest merge.
Am I getting warmer?

gold2718 commented:
My stuff is now working again, thanks for fixing this.

jedwards4b commented:
The doc didn't change, I did change cime_developer.  I added an ERP test and I removed some _Ld3 modifiers that were masking the problem we found in ERR.

rljacob commented:
Does this fix #1006?
